stripe/src/main/java/com/stripe/android/PaymentSession.java

```diff
@@ -0,0 +1,204 @@
+        outState.putParcelable(PAYMENT_SESSION_DATA_KEY, mPaymentSessionData);
+    }
+
+    /**
+     * Set the cart total for this PaymentSession. This should not include shipping costs.
+     *
+     * @param cartTotal the current total price for all non-shipping and non-tax items in
+     *                  a customer's cart
+     */
+    public void setCartTotal(@IntRange(from = 0) long cartTotal) {
+        mPaymentSessionData.setCartTotal(cartTotal);
+    }
+
+    private void fetchCustomer() {
+        if (mPaymentSessionListener != null) {
+            mPaymentSessionListener.onCommunicatingStateChanged(true);
+        }
+        CustomerSession.getInstance().retrieveCurrentCustomer(
+                new CustomerSession.CustomerRetrievalListener() {
+                    @Override
+                    public void onCustomerRetrieved(@NonNull Customer customer) {
+                        String paymentId = customer.getDefaultSource();
+                        mPaymentSessionData.setSelectedPaymentMethodId(paymentId);
+
+                        if (mPaymentSessionListener != null) {
+                            mPaymentSessionListener
+                                    .onPaymentSessionDataChanged(mPaymentSessionData);
+                            mPaymentSessionListener.onCommunicatingStateChanged(false);
+                        }
+                    }
+
+                    @Override
+                    public void onError(int errorCode, @Nullable String errorMessage) {
+                        if (mPaymentSessionListener != null) {
+                            mPaymentSessionListener.onError(errorCode, errorMessage);
+                            mPaymentSessionListener.onCommunicatingStateChanged(false);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Represents a listener for PaymentSession actions, used to update the host activity
+     * when necessary.
+     */
+    public interface PaymentSessionListener {
+        /**
+         * Notification method called when network communication is beginning or ending.
+         *
+         * @param isCommunicating {@code true} if communication is starting, {@code false} if it
+         * is stopping.
+         */
+        void onCommunicatingStateChanged(boolean isCommunicating);
+
+        /**
+         * Notification method called when an error has occurred.
+         *
+         * @param errorCode a network code associated with the error
+         * @param errorMessage a message associated with the error
+         */
+        void onError(int errorCode, @Nullable String errorMessage);
+
+        /**
+         * Notification method called when the {@link PaymentSessionData} for this
+         * session has changed.
+         *
+         * @param data the updated {@link PaymentSessionData}
+         */
+        void onPaymentSessionDataChanged(@NonNull PaymentSessionData data);
```

Comment:
@bdorfman-stripe what are your thoughts on the interface for when a payment finishes?

seems like we could either bundle the resulting `source`/`status`/`error` in `PaymentSessionData`, or create another `PaymentSessionResult` object?